src/lib/FlightTasks/tasks/AutoLineSmoothVel/FlightTaskAutoLineSmoothVel.cpp

```diff
@@ -110,6 +143,14 @@ inline float FlightTaskAutoLineSmoothVel::_constrainOneSide(float val, float con
 	return math::constrain(val, min, max);
 }
 
+void FlightTaskAutoLineSmoothVel::_initEkfResetCounters()
```

Comment:
Why is this method duplicated in `FlightTaskManualPositionSmoothVel`? Should it not be added to the common ancestor that anyways evaluates `_sub_vehicle_local_position`?

Reply:
The common ancestor of Auto and ManualPosition is - if I remember correctly - FlightTask. I agree that this duplication is far from nice, but I don't see how to do it otherwise...

Reply:
In principle the subscription of the vehicle position state is handled here: https://github.com/PX4/Firmware/blob/f613581b701700d2abea1caf132ec17788ce0bed/src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp#L89 and hence the common ancestor makes sense... I'm also not sure how to handle ekf resets properly. I'd like to have them wrapped because resetting every state in every possible position related task is very cumbersome and error-prone.